### PR TITLE
feat: #48 add table with mentoring ways to user page

### PR DESCRIPTION
**Pull request by Howl404**

### CODE REVIEW

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts [hunk 1]
```diff
@@ -79,8 +79,19 @@ export class WayPreviewDAL {
   /**
    * Get User own ways preview
    */
-  public static async getOwnWaysPreview(uuid: string): Promise<WayPreview[]> {
-    const waysDTO = await WayService.getOwnWaysDTO(uuid);
+  public static async getUserWaysPreview(uuid: string, type: "OwnWays" | "MentoringWays"): Promise<WayPreview[]> {
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:83]: Add to comment that you get not only own ways but also mentoring ways

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 2]
```diff
@@ -17,11 +18,22 @@ export const UserPage = () => {
...
+              level={HeadingLevel.h3}
+            />
+            <OwnWaysTable uuid={uuid} />
+            <Title
+              text="Mentoring ways"
+              level={HeadingLevel.h3}
+            />
+            <MentoringWaysTable uuid={uuid} />
+          </>
+        )
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/userPage/UserPage.tsx:35]: I think better move this logic to separate function

File: mw-webapp/src/logic/waysTable/MentoringWaysTable.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+   */
+  uuid: string;
+}
+
+/**
+ * Render table of mentoring ways preview
+ */
+export const MentoringWaysTable = (props: MentoringWaysTableProps) => {
+  const [ownWays, setOwnWays] = useState<WayPreview[]>([]);
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/waysTable/MentoringWaysTable.tsx:23]: If you load mentoringWays why state calls ownWays

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 2 above)
Comment (Howl404) [reply mw-webapp/src/logic/userPage/UserPage.tsx:35]: I think it's ok right now, another function is kinda unnecessary in my opinion

File: mw-webapp/src/logic/waysTable/MentoringWaysTable.tsx (hunk 3 above)
Comment (Howl404) [reply mw-webapp/src/logic/waysTable/MentoringWaysTable.tsx:23]: Already renamed

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts (hunk 1 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:83]: Done

**Review by SergioAJS — APPROVED** [review]

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 2 above)
Comment (tritonJS826) [reply mw-webapp/src/logic/userPage/UserPage.tsx:35]: Personally I don't like ternarian operator is such cases. Also I think that we should not redirect from jsx, better to redirect on the next line after reading  params, it is about encapsulation, let's keep cohesive data and logic nearby

**Review by tritonJS826 — APPROVED** [review]

**Review by Ekaterina1994 — APPROVED** [review]

**Pull request merged**